TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -35,18 +35,17 @@
 
 import java.util.HashMap;
 
-public class BotActivity extends AppCompatActivity implements BotView {
+public class BotActivity extends AppCompatActivity implements BotView, DeviceConnection.DeviceConnectionListener {
```

Comment:
The user only cares about device connections, disconnections and when we receive data, everything else is an implementation detail and should be hidden in the relevant packages behind a public interface.